python/ml_wrappers/model/predictions_wrapper.py

```diff
@@ -47,17 +53,23 @@ def _get_filtered_data(
         :return: The filtered dataframe based on the values in query data.
         :rtype: pd.DataFrame
         """
-        queries = []
-        for column_name, column_data in query_test_data_row.squeeze().items():
-            if isinstance(column_data, str):
-                queries.append("`{}` == '{}'".format(
-                    column_name, column_data))
-            else:
-                queries.append("`{}` == {}".format(
-                    column_name, column_data))
-
-        queries_str = '(' + ') & ('.join(queries) + ')'
-        filtered_df = self._combined_data.query(queries_str)
+        if self._should_construct_pandas_query:
```

Comment:
should the if/else be switched here? isn't this the "not self._should_construct_pandas_query" since the else is actually making the query?

Reply:
Thanks for pointing this out. I changed the if condition.

Thankfully, I had interpreted the timing results correctly ;).